### ISSUE TITLE
subresource_adapter::AspectParameters' has virtual functions but non-virtual destruct

### ISSUE THREAD

**Issue by timvpGoogle** [KhronosGroup/Vulkan-ValidationLayers#2016]
Building the validation layers with `-Wnon-virtual-dtor` gives the following error:

```
external/angle/third_party/vulkan-validation-layers/src/layers/subresource_adapter.h:48:7: error: 'subresource_adapter::AspectParameters' has virtual functions but non-virtual destruct
or [-Werror,-Wnon-virtual-dtor]
class AspectParameters {
      ^
1 error generated.
```

This can be fixed by adding a virtual destructor:

`layers/subresource_adapter.h`
```
class AspectParameters {
  public:
    virtual ~AspectParameters() {}
```

**Comment by timvpGoogle** [comment]
Here's a pull request:
https://github.com/KhronosGroup/Vulkan-ValidationLayers/pull/2017

I don't know how to use github (and barely know git), so let me know where I did something wrong.

**Comment by timvpGoogle** [comment]
#2017

**Comment by mark-lunarg** [comment]
Closed by #2017!
